pkg/k8s/apps/crud_test.go

```diff
@@ -543,3 +544,337 @@ func TestListDevModeOn(t *testing.T) {
+		deployment           *appsv1.Deployment
+		expectedNodeSelector map[string]string
+		shouldInherit        bool
+	}{
+		{
+			name:        "inherit nodeSelector when env var is true and dev nodeSelector is empty",
+			envVarValue: "true",
+			dev: &model.Dev{
+				Name:         "test",
+				Container:    "app",
+				NodeSelector: nil,
+			},
+			deployment: &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test",
+					Namespace: "test",
+				},
+				Spec: appsv1.DeploymentSpec{
+					Template: v1.PodTemplateSpec{
+						Spec: v1.PodSpec{
+							NodeSelector: map[string]string{
+								"kubernetes.io/os":   "linux",
+								"kubernetes.io/arch": "amd64",
+							},
+							Containers: []v1.Container{
+								{
+									Name:  "app",
+									Image: "nginx:latest",
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedNodeSelector: map[string]string{
+				"kubernetes.io/os":   "linux",
+				"kubernetes.io/arch": "amd64",
+			},
+			shouldInherit: true,
+		},
+		{
+			name:        "do not inherit when env var is false",
+			envVarValue: "false",
+			dev: &model.Dev{
+				Name:         "test",
+				Container:    "app",
+				NodeSelector: nil,
+			},
+			deployment: &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test",
+					Namespace: "test",
+				},
+				Spec: appsv1.DeploymentSpec{
+					Template: v1.PodTemplateSpec{
+						Spec: v1.PodSpec{
+							NodeSelector: map[string]string{
+								"node-type": "gpu",
+							},
+							Containers: []v1.Container{
+								{
+									Name:  "app",
+									Image: "nginx:latest",
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedNodeSelector: nil,
+			shouldInherit:        false,
+		},
+		{
+			name:        "do not inherit when dev nodeSelector is not empty",
+			envVarValue: "true",
+			dev: &model.Dev{
+				Name:      "test",
+				Container: "app",
+				NodeSelector: map[string]string{
+					"custom-label": "custom-value",
+				},
+			},
+			deployment: &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test",
+					Namespace: "test",
+				},
+				Spec: appsv1.DeploymentSpec{
+					Template: v1.PodTemplateSpec{
+						Spec: v1.PodSpec{
+							NodeSelector: map[string]string{
+								"kubernetes.io/os": "linux",
+							},
+							Containers: []v1.Container{
+								{
+									Name:  "app",
+									Image: "nginx:latest",
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedNodeSelector: map[string]string{
+				"custom-label": "custom-value",
+			},
+			shouldInherit: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Set environment variable
+			if tt.envVarValue != "" {
+				os.Setenv(model.OktetoInheritKubernetesNodeSelectorEnvVar, tt.envVarValue)
+			} else {
+				os.Unsetenv(model.OktetoInheritKubernetesNodeSelectorEnvVar)
+			}
+			defer os.Unsetenv(model.OktetoInheritKubernetesNodeSelectorEnvVar)
```

Comment:
Let's use t.setenv, so we don't have to explicitly clean it up. Also, let's try to remove the if/else